docs/RingBuffer.md

```diff
@@ -0,0 +1,414 @@
+# New ebpf Ring Buffer Map (proposal)
+
+## Overview
+
+The current ringbuffer uses a pure callback-based approach to reading the ringbuffer.
+Linux also supports memory-mapped polling consumers, which can't be directly supported in the current model.
```

Comment:
What happens if you try to port code to do so?  Would it fail to compile, compile and crash when run, compile and have some other behavior when run, or what?